scripts/pages/zoning/zoning.ts

```diff
@@ -334,15 +336,23 @@ class ZoneMenuHandler {
 
 		if (!this.mapZoneData) return;
 
-		this.panels.leftList.RemoveAndDeleteChildren();
-		this.panels.centerList.RemoveAndDeleteChildren();
-		this.panels.rightList.RemoveAndDeleteChildren();
+		const oldScrollOffset = {};
+
+		// Record the current scroll position. If the panel is overflowing,
+		// this will be used to match the scroll position after updating list panels
+		for (const [position, panel] of Object.entries(this.panels.selectionLists)) {
+			oldScrollOffset[position] = panel?.GetParent()?.scrolloffset_y ?? 0;
+		}
```

Comment:
Would be better to use a name like "listIndex" instead of "position" since "position" sounds like it's the scroll position.

Reply:
I would use Object.values and use the panel IDs as keys. I tend to avoid using numeric keys in Objects in general - it works okay in this case, but the index keys are stringified when inserting and accessing, whilst giving the impression of being numbers, can be a bit hard to understand.

Reply:
@jason-e does it make sense to just call the key "list" since they're strings rather than numbers ('left', 'center', and 'right')?

Reply:
Yeah "list" seems fine

Reply:
Sorry to nitpick again but list is also weird because it sounds like its an array and not a string. Maybe "listKey", or list "listType" etc